zhaquirks/ikea/opencloseremote.py

```diff
@@ -65,7 +65,7 @@ def handle_cluster_request(
         We just want to keep track of direction, to associate it with the stop command.
         """
 
-        cmd_name = self.server_commands.get(hdr.command_id, [hdr.command_id])[0]
+        cmd_name = self.server_commands.get(hdr.command_id, (hdr.command_id)).name
```

Comment:
I think this should be enough, since we shouldn't see any unknown commands:

```suggestion
        cmd_name = self.server_commands[hdr.command_id].name
```

Reply:
Way must **I** taking away so nice working code from the system then "_others_" is adding tones of code with different quality ? ;-(((  !!

Have patching the HA container and its working gear and all DA looks matching with the commands !!


Great work and thanks for the help **J** & **P** !!

Reply:
@MattWestb Did you try the changes puddly mentioned? So just this:
```python
cmd_name = self.server_commands[hdr.command_id].name
```
The remote should only send known ZCL commands. So that could should be enough.

(We should eventually also add some basic tests for this)

Reply:
Yes and all events used in the DA i is getting OK

Reply:
Ok, you can commit/push that change to the PR then